core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -0,0 +1,227 @@
+        this.authorisedOps = authorisedOps;
+    }
+
+    public LinkedHashMap<String, AdditionalOperations> getAuthorisedOps() {
+        return authorisedOps;
+    }
+
+    /**
+     * Adds in the additional Operations specified.  The original opChain will
+     * be updated.
+     *
+     * @param opChain the {@link OperationChain} being executed.
+     * @param user    the {@link User} executing the operation chain
+     */
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final User user) {
+        final List<Operation> newOpList = new ArrayList<>();
+
+        boolean hasAuth = false;
+        if (!authorisedOps.isEmpty() && !user.getOpAuths().isEmpty()) {
+            for (final String auth : authorisedOps.keySet()) {
+                if (user.getOpAuths().contains(auth)) {
+                    newOpList.addAll(addOperationsToChain(opChain, authorisedOps.get(auth)));
+                    hasAuth = true;
+                    break;
+                }
+            }
+        }
+
+        if (!hasAuth) {
+            newOpList.addAll(addOperationsToChain(opChain, defaultOperations));
+        }
+
+        opChain.getOperations().clear();
+        opChain.getOperations().addAll(newOpList);
+    }
+
+    @Override
+    public <T> T postExecute(final T result,
+                             final OperationChain<?> opChain, final User user) {
+        return result;
+    }
+
+    private static AddOperationsToChain fromJson(final byte[] json) {
+        try {
+            return JSON_SERIALISER.deserialise(json, AddOperationsToChain.class);
+        } catch (final SerialisationException e) {
+            throw new SchemaException("Failed to load element definitions from bytes", e);
+        }
+    }
+
+    private static AddOperationsToChain fromJson(final InputStream inputStream) {
+        try {
+            return JSON_SERIALISER.deserialise(inputStream, AddOperationsToChain.class);
+        } catch (final SerialisationException e) {
+            throw new SchemaException("Failed to load element definitions from bytes", e);
+        }
+    }
+
+    private void setOperations(final AddOperationsToChain addOperations) {
+        if (addOperations.getAuthorisedOps() != null) {
+            this.setAuthorisedOps(addOperations.getAuthorisedOps());
+        }
+        addOperations.defaultOperations.getStart();
+        if (addOperations.defaultOperations.getStart() != null) {
+            this.defaultOperations.setStart(addOperations.defaultOperations.getStart());
+        }
+        if (addOperations.defaultOperations.getEnd() != null) {
+            this.defaultOperations.setEnd(addOperations.defaultOperations.getEnd());
+        }
+        if (addOperations.defaultOperations.getBefore() != null) {
+            this.defaultOperations.setBefore(addOperations.defaultOperations.getBefore());
+        }
+        if (addOperations.defaultOperations.getAfter() != null) {
```

Comment:
I think these null checks could go in the setters in the AdditionalOperations class instead.